Refactor logging functions to use %s formatting instead of f-strings

StarKBhaviN opened the pull request:
Fixes #3820

You can use % formatting but leave interpolation to the logging function by passing the parameters as arguments.
If logging-not-lazy is disabled then you can use % formatting as normal.

suhaibmujahid reviewed (commented):
Thank you for the PR, @StarKBhaviN! Could you please fix the failing tests?

Using pre-commit will help in fixing the code style errors: https://github.com/mozilla/bugbug#auto-formatting

StarKBhaviN commented:
[image: image.png]

Why i am getting this error

On Sun, Nov 19, 2023 at 12:14 PM Suhaib Mujahid ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
>
> Thank you for the PR, @StarKBhaviN <https://github.com/StarKBhaviN>!
> Could you please fix the failing tests?
>
> Using pre-commit will help in fixing the code style errors:
> https://github.com/mozilla/bugbug#auto-formatting
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/mozilla/bugbug/pull/3822#pullrequestreview-1738550555>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AZKBUNAEG3SDEVC2CD7QKODYFGTF3AVCNFSM6AAAAAA7RP2HZ2VHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMYTOMZYGU2TANJVGU>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

suhaibmujahid commented on .flake8:
Unrelated change

StarKBhaviN replied:
This change made automatically when running command...

- pre-commit run --all-files

suhaibmujahid replied:
This is due to the changes in the config file (i.e., `.pre-commit-config.yaml`). All unrelated changes should be dropped.

suhaibmujahid commented on .pre-commit-config.yaml:
Unrelated changes

StarKBhaviN replied:
This change made automatically when running command...

- pre-commit install

suhaibmujahid replied:
It seems that the file was modified by mistake. The `pre-commit install` command should not modify the config file. Modifying the config file caused the other modification.

suhaibmujahid commented on bugbug/labels/regression_bug_nobug.csv:
Unrelated changes

StarKBhaviN replied:
This change made automatically when running command...

- pre-commit run --all-files

suhaibmujahid replied:
You do not need to execute `pre-commit run --all-files`; once you have pre-commit installed (i.e., `pre-commit install`) pre-commit will run automatically when you commit. If you want to run it manually, using `pre-commit run` should be enough to lint the modified files.

suhaibmujahid commented on http_service/.dockerignore:
Unrelated change

StarKBhaviN replied:
This change made automatically when running command...

- pre-commit run --all-files

suhaibmujahid commented on scripts/regressor_finder.py:
```diff
@@ -500,7 +503,7 @@ def results():
                     if exc is not None:
                         logger.info(
-                            f"Exception {exc} while analyzing {futures[future]}"
+                            "Exception %d while analyzing %s", exc, futures[future]
```
The value of `exc` is an Exception, not an integer.

StarKBhaviN replied:
Yeah fixed that!!!

suhaibmujahid commented on scripts/trainer_extract_args.py:
```diff
@@ -11,7 +11,7 @@
 
 
-def get_model_name() -> str | None:
+def get_model_name() -> str or None:
```
Unrelated change

StarKBhaviN replied:
Yeah fixed it

suhaibmujahid commented on tests/fixtures/bug_features/nightly_uplift.json:
Unrelated change

StarKBhaviN replied:
This change made automatically when running command...

- pre-commit run --all-files

suhaibmujahid commented on tests/fixtures/bugs.json:
Unrelated change

suhaibmujahid commented on bugbug/models/regressor.py:
```diff
@@ -278,7 +278,7 @@ def evaluation(self) -> None:
         total_regressions = sum(1 for _, is_reg in results if is_reg)
         average_regression_rate = total_regressions / total_landings
 
-        logger.info("Average risk is %d", average_regression_rate)
```
This is a valid fix but unrelated to the goal of this PR. We could open a new issue and fix it in a separate PR (with similar cases if there are any)

StarKBhaviN replied:
Okay got it

suhaibmujahid replied:
I filed a separate issue for this (i.e., #3837). This fix is not related to this PR. Please revert it.

suhaibmujahid reviewed (commented):
Thank you, @StarKBhaviN! Could you please take care of the following:
- Drop unrelated changes
- Link the PR to the issue that it is fixing (in the PR description[^1])
- Edit the PR title to clarify that this is related to logging  

[^1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue

StarKBhaviN commented:
![image](https://github.com/mozilla/bugbug/assets/106175028/c37f1e32-3420-476a-b02e-ff5111768382)

This 3 errors does not letting me commit.

Issues resolved :
- Reverted commit eed0ef2e89d57fa9d0c19c5a6a289b51ad8418dc
- Changed only required code.
- Now trying to commit i am getting error. HELP!

jpangas commented:
@StarKBhaviN , Bugbug requires Python 3.10+ as mentioned in the [Setup and Prerequisities](https://github.com/mozilla/bugbug#setup-and-prerequisites) 

> Currently, Python 3.10+ is required. You can double check the version we use by looking at setup.py.

You're using Python `3.9.13 `which is not supported. I would advise you create a virtual environment using `venv` or `conda` for `bugbug` and its dependencies. Remember to use Python 3.10+

StarKBhaviN commented:
![image](https://github.com/mozilla/bugbug/assets/106175028/53cd45c0-bc19-4463-9d23-722cbb48ead9)
Still getting same error after creating virtual env and reinstalling dependencies

jpangas commented:
Let's chat in the bugbug matrix room. It will be easier to follow up from there and others will be available to help.

suhaibmujahid commented:
>  Still getting same error after creating virtual env and reinstalling dependencies

Try to re-install `pre-commit` after activating venv:
```
pre-commit uninstall
pre-commit install
```

StarKBhaviN commented:
![image](https://github.com/mozilla/bugbug/assets/106175028/5d63f886-a506-457f-8a76-009cd6b7fafc)

bugbug lint is failing in codespell

suhaibmujahid commented:
> bugbug lint is failing in codespell

Because the PR still changes in the `.pre-commit-config.yaml` file. All unrelated changes should be dropped. Please just keep the changes related to the logging statements mentioned in the issue.

suhaibmujahid reviewed (commented):
Thank you @StarKBhaviN! The changes to the following files are not related to this PR. Please revert the changes:
- .flake8
- bugbug/labels/regression_bug_nobug.csv
- http_service/.dockerignore
- tests/fixtures/bug_features/nightly_uplift.json
- tests/fixtures/bugs.json

suhaibmujahid commented on scripts/commit_classifier.py:
```diff
@@ -294,7 +294,7 @@ def apply_phab(self, hg, phabricator_deployment, diff_id):
             if self.has_revision(hg, patch.base_revision):
                 logger.info(
-                    f"Stopping at diff {patch.id} and revision {patch.base_revision}"
+                    "Stopping at diff %f and revision %f", patch.id, patch.base_revision
```
Could you clarify why `%f` is used for `patch.id`?

StarKBhaviN replied:
It should not be %f.
I am confused between %d and %s, so now i am using .format method

suhaibmujahid commented on scripts/commit_classifier.py:
```diff
@@ -383,7 +383,7 @@ def load_user(phid):
 
             logger.info(
-                f"Applying {patch.phid} from revision {revision['id']}: {message}"
+                "Applying %f from revision {revision['id']}: %s", patch.phid, message
```
The revision id here is missing.

suhaibmujahid commented on scripts/maintenance_effectiveness_indicator.py:
```diff
@@ -54,7 +54,7 @@ def main() -> None:
 
     for factor, value in result.items():
-        logger.info(f"{factor}: {round(value, 2) if value != math.inf else value}")
+        logger.info("%s : %d", factor, round(value, 2) if value != math.inf else value)
```
The whitespace was not present in the original login statement:
```suggestion
        logger.info("%s: %d", factor, round(value, 2) if value != math.inf else value)
```

suhaibmujahid commented on scripts/regressor_finder.py:
```diff
@@ -608,11 +617,11 @@ def evaluate(bug_introducing_commits):
     logger.info("Found %d regressors out of %d", found_regressors, all_regressors)
     logger.info(
-        f"Misassigned {misassigned_regressors} regressors out of {all_regressors}"
+        "Misassigned %d regressors out of %d", {misassigned_regressors}, all_regressors
```
```suggestion
        "Misassigned %d regressors out of %d", misassigned_regressors, all_regressors
```

suhaibmujahid commented on scripts/commit_classifier.py:
```diff
@@ -294,7 +294,9 @@ def apply_phab(self, hg, phabricator_deployment, diff_id):
             if self.has_revision(hg, patch.base_revision):
                 logger.info(
-                    f"Stopping at diff {patch.id} and revision {patch.base_revision}"
+                    "Stopping at diff {} and revision {}".format(
+                        patch.id, patch.base_revision
+                    )
```
The goal of the issue is to move from "f-string" to "%s formatting", not to `str.format`. See the reason here: https://pylint.readthedocs.io/en/latest/user_guide/messages/warning/logging-fstring-interpolation.html

If you are in doubt about what the type is, you could execute the script to check. If it is hard to determine, `%s` should be okay in most cases.

suhaibmujahid commented on scripts/commit_classifier.py:
```diff
@@ -383,7 +383,7 @@ def load_user(phid):
 
             logger.info(
-                f"Applying {patch.phid} from revision {revision['id']}: {message}"
+                "Applying %s from revision %s: %s", patch.phid, message, revision["id"]
```
The order here is not correct:
```suggestion
                "Applying %s from revision %s: %s", patch.phid, revision["id"], message
```

suhaibmujahid reviewed (changes requested):
Please rename the PR to something more descriptive (e.g., “Use %s formatting instead of f-string in logging functions”). Also, could you please link to the issue in the PR description?[^1]

[^1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue

suhaibmujahid commented on scripts/regressor_finder.py:
```diff
@@ -340,7 +344,7 @@ def mercurial_to_git(revs):
         )
         logger.info(
-            f"Already classified {len(prev_bug_introducing_commits)} commits..."
+            "Already classified %s commits...", len(prev_bug_introducing_commits)
```
The value here is an integer:
```suggestion
            "Already classified %d commits...", len(prev_bug_introducing_commits)
```

suhaibmujahid reviewed (approved):
LGTM! Thank you, @StarKBhaviN!
